src/lib/remark-plugin-adjust-link-urls.ts

```diff
@@ -1,15 +1,57 @@
+import path from 'path'
 import { visit } from 'unist-util-visit'
 import { Link } from 'mdast'
 import { Plugin, Transformer } from 'unified'
 
+const getIsInternalPath = (url: string) => {
+	try {
+		new URL(url)
+		return false
+	} catch (e) {
+		return true
+	}
+}
+
 const remarkPluginAdjustLinkUrls: Plugin = ({
+	currentPath = '',
 	urlAdjustFn,
 }: {
+	currentPath: string
 	urlAdjustFn: (url: string) => string
 }): Transformer => {
 	return function transformer(tree) {
 		visit(tree, ['link', 'definition'], (node: Link) => {
-			node.url = urlAdjustFn(node.url)
+			let urlToAdjust = node.url
+
+			/**
+			 * This is to handle links like "variables/input", which are effectively
+			 * the same thing as "./input" in the context of its current page.
```

Comment:
I'm not actually sure this is correct, but taking at shot at expanding the description here since it wasn't immediately clear to me what the `!/^[./?#]/.test(node.url)` regex was intended for either, or why `currentPath` was needed.

```suggestion
			 * the same thing as "./input" in the context of its current page.
			 *
			 * For context, links that do _not_ start with `.`, `/`, `?`, or `#`,
			 * such as "variables/input", are intended to be relative links.
			 *
			 * These links need to be prefixed with the `currentPath`, 
			 * as `next/link` will otherwise treat relative links as relative to the
			 * current page file, rather than relative to the current page URL.
			 * But, we're also not sure if the path parts in the relative link might
			 * overlap with path parts in the currentPath, so we iterate over
			 * `currentPathParts` until we find overlap, and use the non-overlapping
			 * parts to prefix the relative link.
```